### ISSUE TITLE
Update fund and location codes in Finance & Purchasing Portal

### ISSUE THREAD

**Issue by atdservicebot** [cityofaustin/atd-data-tech#17947]
> What application are you using?

Finance & Purchasing Portal

> Describe the problem.

Fund 7790 needs to be added. Office Location Codes are also outdated. Please add Ben White, OTC, Townview, St. Elmo, Dalton, Kramer.

> Describe the outcome you'd like to see when this feature is implemented.

A uniform system that provides all Fund codes and office location codes that match what we have on TPW SharePoint.

> Describe any workarounds you currently have in place or alternative solutions you've considered.

A LOT of Notes to Purchasing or simply shipping to my office since it is currently listed.

> Is there anything else we should know?

Before Barton Oaks is vacated on 08/16.

> Requested By
Jeff S.

Request ID: DTS24-115715

**Comment by stephaniemoorer** [comment]
Contacted Jeff with some additional questions:
Are you referring to the Office Location Code field that is located on the New Purchase Request form?
Yes. I assume many of the codes serve a purpose, but none of the legacy PWD and CPO facilities have been added to the dropdown since our merger and reorg in March 2023. If possible, it would be nice to have them match the Locations on the hover menu of [TPW Hub](https://cityofaustin.sharepoint.com/sites/TPWHub?xsdata=MDV8MDJ8U3RlcGhhbmllLk1vb3JlcjJAYXVzdGludGV4YXMuZ292fDZiNjViZmIwY2QxMDRhODI2YWIyMDhkYzkyMTUyNTA3fDVjNWUxOWY2YTZhYjRiNDViMWQwYmU0NjA4YTlhNjdmfDB8MHw2Mzg1NDU4NjY3MTE1MDY1NTR8VW5rbm93bnxUV0ZwYkdac2IzZDhleUpXSWpvaU1DNHdMakF3TURBaUxDSlFJam9pVjJsdU16SWlMQ0pCVGlJNklrMWhhV3dpTENKWFZDSTZNbjA9fDB8fHw%3d&sdata=cGpuMVhESk52UFhuTnA1b3lCcDZZWWlLVWpBOVgwaWVqWTluOEd5eG45VT0%3d).
Can you provide the link to the SharePoint location that has all the Fund Codes and Location Codes you would like to have displayed on the Finance Portal?
Sure! The Locations are at [https://cityofaustin.sharepoint.com/sites/TPWHub](https://cityofaustin.sharepoint.com/sites/TPWHub?xsdata=MDV8MDJ8U3RlcGhhbmllLk1vb3JlcjJAYXVzdGludGV4YXMuZ292fDZiNjViZmIwY2QxMDRhODI2YWIyMDhkYzkyMTUyNTA3fDVjNWUxOWY2YTZhYjRiNDViMWQwYmU0NjA4YTlhNjdmfDB8MHw2Mzg1NDU4NjY3MTE1MTkwMDF8VW5rbm93bnxUV0ZwYkdac2IzZDhleUpXSWpvaU1DNHdMakF3TURBaUxDSlFJam9pVjJsdU16SWlMQ0pCVGlJNklrMWhhV3dpTENKWFZDSTZNbjA9fDB8fHw%3d&sdata=VnJNSGJpMTNDb2xjakNQY2VIN3pjbmxrakpnUUZURFpZYk5wYjlIb08wYz0%3d). TPW's Fund, Department and Unit (FDU) codes are in the attached email from Jenny King. I know Fund 7790 - Child Safety is not listed in Knack. Several Department and Unit codes are also missing. This creates more work for the teams that joined ATD through the merger and creates an "othering" effect.

From the information provided it looks like the following Fund Codes and Locations are not listed:
1004 - General Fund Budget Stabilization Reserve Fund
7790 - Child Safety Fund
6221 - Bicycle Projects 00
6222 - Pedestrian Projects FY00
6242 - Disaster Land Acquisition
6KUA - East Austin Sidewalks
Locations:
Rio - Rio Grande
OTC - Barton Springs
Ben White
Townview
St Elmo
Dalton
Kramer

**Comment by stephaniemoorer** [comment]
@dianamartin can you give me some directions on how to get this information updated?  Do we need to get approval from someone in Finance? Last time I emailed Tim T for approval to add a FDU it went around in circles.

**Comment by dianamartin** [comment]
@stephaniemoorer I can add these if we need to

**Comment by stephaniemoorer** [comment]
@dianamartin still searching for where the location codes came from.  Peggy suggested that at one time Banner had location codes. I know we get employee data from Banner bi weekly do you have a way to check if a location code is on the report?

**Comment by dianamartin** [comment]
@stephaniemoorer I do rememeber the Banner feed does have a location code. cc @ChristinaTremel, are we getting it today?

**Comment by dianamartin** [comment]
@stephaniemoorer what's the status or next steps we want to do for this?

**Comment by stephaniemoorer** [comment]
@dianamartin 

I would like to clean up the list to look more like the Office Location list used in the HR Portal
![{F38D24FE-2459-428A-9DEB-D8D6F3B982B8}](https://github.com/user-attachments/assets/7cf81159-6ed4-4007-85a8-8b36bff27bac)

- Guess we would need to discuss with someone in Finance before making any changes.

In the meantime the current Office Location list in Purchase Request could be updated:

- [ ] ATDWHSE - should this be TPWWHSE (is WHSE for warehouse?)
- [ ] ATD Marking Section - should this be TPW Marking Section
- [ ] COA Transportation / Barton Oaks - can this be deleted since TPW moved from Barton Oaks?
- [ ] COA Transportation / 8900 - should this be 8900 Cameron to match 8700 Cameron?


![{9EC4C91E-7808-4909-A5BC-D1876610D9C6}](https://github.com/user-attachments/assets/0a4b7537-e561-4b7d-8385-b32c477fea9e)

**Comment by stephaniemoorer** [comment]
@dianamartin
I contacted the AIMSHelp email to ask for a report of shipping locations for TPW.
AIMSHelp response:

We do not have an existing report to pull all that have been created.  I searched using the department number and found none, but a department number isn't required on the address.  Most addresses don't have a dept. # associated with them.  My supervisor is creating a report where we can pull all of them to review.
In the meantime, are there some addresses you need to verify we have?

- I sent them the list of locations that don't have a location code. Hopefully they have the info we need. 
- I also requested the full report of locations - not sure if they will provide that, it's probably massive

**Comment by dianamartin** [comment]
# 9/18/24
List from spreadsheet from AIMS

Location Name | Code | Notes
-- | -- | -- | 
COA Transportation/8700 Cameron | TP87 | Same as Finance and Purchasing List, No action.
COA Transportation/8900 Cameron | TP89 | Same as Finance and Purchasing List, No action.
COA Transportation/Ben White 450 | TP90 | Need to add	
COA Transportation/Toomey | TPT | Same as Finance and Purchasing List, No action.
Austin Code Department/Ben White 550 | ACD |Need to request a new location under TPW, ask Finance?  
Public Works Street & Bridge-Dalton Lane Yard | TP01 | Need to request a new location under TPW, ask Finance?  
Public Works Parking/Rio Grande | PWP | Need to request a new location AND code under TPW, ask Finance?  
Public Works - Meinardus | PWM | Need to request a new location under TPW, ask Finance?  
Public Works/Townview Yard | TP9 | Need to request a new location under TPW, ask Finance?  
Public Works Dept/Barton Springs - OTC | TP21 | Need to request a new location under TPW, ask Finance?

**Comment by stephaniemoorer** [comment]
10/1/24
Provided Jenny with the Shipping Location spreadsheet from AIMS.
she will review and provide list of updates to AIMS Help, we will then update in Knack

**Comment by dianamartin** [comment]
# 12/13/24
Can we ping Jenny to get this list back to us? Jeff is hoping to get his updated by the new year.

**Comment by stephaniemoorer** [comment]
12/13/24
emailed Jenny and Jon for a update

**Comment by stephaniemoorer** [comment]
## 1/13/2025
During our meeting about updating the PR Quiz I asked Jon for update. 
He indicated he has an updated list. He will forward the list to me.

**Comment by stephaniemoorer** [comment]
## 1/27/25
@dianamartin 
Here are the updated  location codes Jon provided for the New Purchase Request Office Location Codes dropdown list

![Image](https://github.com/user-attachments/assets/e2328ca9-73c3-4f95-8669-786dbc8dd7d6)

**Comment by stephaniemoorer** [comment]
## 1/28/25
Diana had some questions regarding several locations
Sent Jon a Teams message with Diana's comments

![Image](https://github.com/user-attachments/assets/d7430e64-15b8-4f03-890c-9d6aa2f0de8e)

Jon's reply:

Hi Stephanie, we can delete tp10 
tp13 i think SBO also has a Markings crew. Prefer to leave it

**Comment by stephaniemoorer** [comment]
## 1/29/25
Confirmed with Jon TP10 and TP13 can be removed from the list
All other location updates can be added

**Comment by dianamartin** [comment]
# 1/30/25
- Was updating the funding codes in the system
- Synced with Jon because I noticed that there were other Office Codes that were dropped. I wanted to make sure the purpose of the codes were for shipment, we discussed taking out the division names worked best. 
  - ex. ATSD was using a code for an old LCRA location from years ago
- [x] Created new field "Location Code" - Multichoice
- [x] Created new field "Office Code" - Short Text
  - [x] Set conditional rule - to aggregate values from the historical field (Office Location Code) and the new (Location Code) to show in the "Office Code" field. This will allow us to keep historical record of what was used in the past. 
![Image](https://github.com/user-attachments/assets/240686bd-37f1-48a8-b746-44356a82b076)

## Updated New PR form
- [x] Removed "Office Location Code"
- [x] Added "Location Code"

![Image](https://github.com/user-attachments/assets/ffe44f0d-aa0f-480a-bdda-a1cbccdf04fc)

## Updated PR Details page
- [x] Added "Office Code" (new field)
>will remove the old field "Office Location Code" once the conditional rule populates the 10k records
![Image](https://github.com/user-attachments/assets/f541b71d-10a1-4899-93f1-a918ae60ccd7)

**Comment by dianamartin** [comment]
# 1/31/25
Messaged Jeff S. (original requester)
>I know you submitted a request awhile ago about updating the TPW location codes in the PR system, I just wanted to let you know that this has been updated. We had to wait on Finance to get these codes updated in AIMs. Thanks for your patience. Hope you have a great weekend!

**Comment by stephaniemoorer** [comment]
## 2/13/25
Have not received any comments or feedback from Jeff S. the original requestor.
Closing this issue
